Третья лабораторная работа выполнена

PopovMatvey opened the pull request:
Третья лабораторная работа выполнена:
1) Выполнены все задания от 1 до 7;
2) В заданиях 1 - 7 выполнены линтеры;
3) В заданиях 1 - 7 выполнены все тесты.

PopovMatvey committed:
Третья лабораторная работа выполнена

Третья лабораторная работа выполнена:
1) Выполнены все задания от 1 до 7;
2) В заданиях 1 - 7 выполнены линтеры;
3) В заданиях 1 - 7 выполнены все тесты.

djilua1 commented on src/ex3_js-objects-part1/task-02.js:
```diff
@@ -0,0 +1,11 @@
+function getPairKeyValue(object) {
+  const PROCESSED_OBJECT = Object.entries(object);
```
Почему UPPER_CASE?

djilua1 commented on src/ex3_js-objects-part1/task-02.js:
```diff
@@ -0,0 +1,11 @@
+    console.log(value);
+  });
+
+  return '';
```
`return undefined;`. Так строка возвращается, а не ничего.

djilua1 commented on src/ex3_js-objects-part1/task-03.js:
```diff
@@ -0,0 +1,18 @@
+function checkONProperty(string, object) {
+  const OBJECT_KEYS = Object.keys(object);
```
Тот же вопрос.

djilua1 commented on src/ex3_js-objects-part1/task-03.js:
```diff
@@ -0,0 +1,18 @@
+  }
+
+  return false;
+}
```
Очень сложно, можно намного проще, поищи метод подходящий.

djilua1 commented on src/ex3_js-objects-part1/task-04.js:
```diff
@@ -0,0 +1,20 @@
+  object[string] = 'new'; // eslint-disable-line no-param-reassign
+
+  return object;
+}
```
Тоже слишком сложно.

djilua1 commented on src/ex3_js-objects-part1/task-05.js:
```diff
@@ -0,0 +1,9 @@
+
+  Object.assign(RETURNED_OBJECT, object);
+
+  return RETURNED_OBJECT;
```
`return Object.assign({}, object);`

djilua1 commented on src/ex3_js-objects-part1/task-06.js:
```diff
@@ -0,0 +1,81 @@
+  }
+
+  return undefined;
+}
```
Слишком сложно, даже не смотрел. Тут можно маленькой функцией обойтись.

djilua1 commented on src/ex3_js-objects-part1/task-07.js:
```diff
@@ -0,0 +1,25 @@
+    Array: () => object.map((i) => getDeepClone(i)),
+  };
+
+  if (object.constructor.name in functionsObject) {
```
Ну уж нет, давай полегче, просто на массив проверь.

PopovMatvey committed:
Изменения по третей лабе

Выполнены изменения по третей лабораторной работе:
1) Во втором задании изменил UPPER_CASE нотацию на CamelCase;
2) Во втором задании изменил return ""; на return undefined;
3) В третьем задании изменил UPPER_CASE нотацию на CamelCase;
4) Также в третьем задании упростил реализацию;
5) В четвёртом задании упростил реализацию;
6) В пятом задании изменил содержимое функции на return Object.assign({}, object);
7) В шестом задании упростил реализацию;
8) В седьмом задании упростил реализацию;
